### ISSUE TITLE
Angular2 CLI

### ISSUE THREAD

**Issue by tapas4java** [ngCli/ng-cli#5]
Hi,

Is this the official Angular2 CLI Repo in which angular team is working/planning now?

I am curious to know the reason behind moving from GULP to BROCCOLI. Angular2 is also moving towards broccoli but not sure why?

**Comment by thetutlage** [comment]
no it is not the official build tool

broccoli is meant to the embedded which means any tool can integrate it with its own build system without any hacks

whereas gulp is a standalone build tool ,gulpfile.js needs to be in project's root and commands should be run from the root itself

**Comment by tapas4java** [comment]
I am planning for creating a Angular2, New Router, Typescript, ES6, Angular Material, SASS fullstack yeomen generator. Do you have any plan to embrace these libraries anytime soon?

If required I can help you on this.

**Comment by thetutlage** [comment]
ngcli is under active development , and i have plans to upgrade it to Angular2.0 but that will take couple of months , as i need to write more examples of Angular2.0.

**Comment by tapas4java** [comment]
Let me know if you need any help in any area. It would be better if we follow JohnPapa's Style guide for scafolding.

If you can explain me how ng-cli works, then i would be able to give you a hand.

**Comment by thetutlage** [comment]
Can you share JohnPapa's style guide , let me see what it offers.

ng-cli itself uses webpack and broccoli to manage the build process , Apart from that it has a option to add your own build process using npm modules and mention them as addons inside package.json file. 

Browsing code will give you some hint , as the code itself is not big

**Comment by tapas4java** [comment]
Here is the link:

https://github.com/johnpapa/angular-styleguide

**Comment by thetutlage** [comment]
That's related to Angular current versions , as of 2.0 all we have is components

**Comment by tapas4java** [comment]
True, but they will treat this as a official style guide and provide migration path based on that.

For now we can play around on alpha releases. Typescript, Angular2,  Angular Material, New Router, all are available on alpha.

**Comment by thetutlage** [comment]
Okay make sense, yes i am writing different examples in order to understand the ideal use case. Any examples from your side will be helpful.

**Comment by tapas4java** [comment]
I have a seed project including Angular 1.4, Material, New Router, ES6, Gulp. 

I will update it to use Angular2 with typescript and let you know. Thanks!
